### PR TITLE
Add websocket-proxy support

### PR DESCRIPTION
**Pull request by cjellick**

### CODE REVIEW

File: code/implementation/host-api/src/main/java/io/cattle/host/api/proxy/launch/ProxyLauncher.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    private static final Logger log = LoggerFactory.getLogger(ProxyLauncher.class);
+
+    @Inject
+    HostApiService hostApiService;
+
+    @Inject
+    ScheduledExecutorService executor;
+
+    @Inject
+    LockDelegator lockDelegator;
```
Comment (ibuildthecloud) [inline code/implementation/host-api/src/main/java/io/cattle/host/api/proxy/launch/ProxyLauncher.java]: not used?

Comment (ibuildthecloud) [inline code/implementation/host-api/src/main/java/io/cattle/host/api/proxy/launch/ProxyLauncher.java]: Not needed

File: code/implementation/host-api/src/main/java/io/cattle/host/api/proxy/launch/ProxyLauncher.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            throw new RuntimeException("Couldn't get public key for websocket-proxy.");
+        }
+        env.put("PATH", System.getenv("PATH"));
+        env.put("PROXY_JWT_PUBLIC_KEY_CONTENTS", pubKey);
+        env.put("PROXY_LISTEN_ADDRESS", ":" + getProxyPort());
+
+        String cattleProxyAddress = "localhost:" + getProxiedPort();
+        env.put("PROXY_CATTLE_ADDRESS", cattleProxyAddress);
+
+        String processName = ManagementFactory.getRuntimeMXBean().getName();
```
Comment (ibuildthecloud) [inline code/implementation/host-api/src/main/java/io/cattle/host/api/proxy/launch/ProxyLauncher.java]: Could also do system("echo $PPID")

**Comment by cjellick** [comment]
For the record @ibuildthecloud verbally LGTMd

**Pull request merged**